addons/actions/src/preview/withActions.ts

```diff
@@ -47,8 +50,25 @@ export const createDecorator = (actionsFn: any) => (...args: any[]) => (storyFn:
     }
     return undefined;
   }, [root, actionsFn, args]);
+}
 
+export const createDecorator = (actionsFn: any) => (...args: any[]) => (storyFn: () => any) => {
+  applyEventHandlers(actionsFn, ...args)
+  
   return storyFn();
 };
 
-export const withActions = createDecorator(actions);
+export const withActions = makeDecorator({
+  name: 'withActions',
+  parameterName: PARAM_KEY,
+  skipIfNoParametersOrOptions: true,
+  allowDeprecatedUsage: false,
+  wrapper: (getStory, context, { parameters }) => {
+    // allow a shortcut of providing just an array.
+    // Anticipating that we'll soon kill configureActions in favor of configuration via parameters
+    const storyOptions = Array.isArray(parameters) ? { handles: parameters } : parameters;
+    applyEventHandlers(actions, ...storyOptions.handles)
+
+    return getStory(context);
+  }
+});
```

Comment:
Curious on folks thoughts on allowing the shortcut syntax. In general we think of parameters as being objects (and our typescript definitions things so to). We'll want to support configuration via parameters too hence not just supporting the array syntax.

Reply:
I'm unsure about it. I definitely get where it is coming from, parameter syntax is verbose and it is a pain.

That said the problem with using a non-object is that it makes it harder to do combinations (say you want to set half the handlers at the component level and the the other half at the story level). That said that'll be a problem with an array at any level.

Probably I would say it is OK as long as there is a more verbose version so if people want to do combining (somehow) they still can. Say for instance in the future we had other keys on the `action` parameter that you wanted to use.

Reply:
ok I removed it. We're going to want combinations for config via params